src/renderer/src/routes/Onboarding/CreateProjectScreen.tsx

```diff
@@ -0,0 +1,282 @@
+	},
+	advancedProjectSettings: {
+		id: 'screens.ProjectCreationScreen.advancedProjectSettings',
+		defaultMessage: 'Advanced Project Settings',
+	},
+	importConfig: {
+		id: 'screens.ProjectCreationScreen.importConfig',
+		defaultMessage: 'Import Config',
+	},
+	errorSavingProjectName: {
+		id: 'screens.ProjectCreationScreen.errorSavingProjectName',
+		defaultMessage:
+			'An error occurred while saving your project name. Please try again.',
+	},
+	saving: {
+		id: 'screens.ProjectCreationScreen.saving',
+		defaultMessage: 'Saving...',
+	},
+})
+
+export const Route = createFileRoute('/Onboarding/CreateProjectScreen')({
+	component: CreateProjectScreenComponent,
+})
+
+const StyledImage = styled('img')({
+	width: 60,
+	height: 60,
+})
+
+const InputWrapper = styled('div')({
+	marginTop: 12,
+	marginBottom: 12,
+	display: 'flex',
+	flexDirection: 'column',
+	alignItems: 'center',
+})
+
+const StyledTextField = styled(TextField)({
+	width: '100%',
+	maxWidth: 400,
+	backgroundColor: WHITE,
+	marginTop: 12,
+})
+
+const CharacterCount = styled(Text, {
+	shouldForwardProp: (prop) => prop !== 'error',
+})<{ error: boolean }>(({ error }) => ({
+	marginTop: 12,
+	color: error ? RED : BLACK,
+	width: '100%',
+	maxWidth: 400,
+	textAlign: 'right',
+}))
+
+const HorizontalLine = styled('div')({
+	borderBottom: `1px solid ${BLUE_GREY}`,
+	margin: '20px auto 20px auto',
+	width: '65%',
+})
+
+function CreateProjectScreenComponent() {
+	const navigate = useNavigate()
+	const { formatMessage } = useIntl()
+	const [projectName, setProjectName] = useState('')
+	const [error, setError] = useState(false)
+	const [errorMessage, setErrorMessage] = useState('')
+	const setProjectNameMutation = useCreateProject()
+
+	const [advancedSettingOpen, setAdvancedSettingOpen] = useState(false)
+	const [configFileName, setConfigFileName] = useState<string | null>(null)
+
+	const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+		const value = event.target.value
+		const localError: boolean = checkForError(
+			value.trim(),
+			PROJECT_NAME_MAX_LENGTH_GRAPHEMES,
+		)
+		setProjectName(value)
+		if (localError !== error) {
+			setError(localError)
+		}
+		setError(localError)
+	}
+
+	const graphemeCount = countGraphemes(projectName.trim())
+
+	const handleAddName = () => {
+		if (checkForError(projectName, PROJECT_NAME_MAX_LENGTH_GRAPHEMES)) {
+			setError(true)
+			return
+		}
+		setProjectNameMutation.mutate(projectName, {
+			onSuccess: () => {
+				navigate({ to: '/tab1' })
+			},
+			onError: (error) => {
+				console.error('Error setting project name:', error)
+				setErrorMessage(formatMessage(m.errorSavingProjectName))
+			},
+		})
+	}
+
+	function importConfigFile() {
+		// Placeholder for file import logic
+		setConfigFileName('myProjectConfig.comapeocat')
+	}
+
+	const topMenu = (
+		<OnboardingTopMenu
+			currentStep={3}
+			onBackPress={() =>
+				navigate({ to: '/Onboarding/CreateJoinProjectScreen' })
+			}
+		/>
```

Comment:
we should also disable the back button when `setProjectNameMutation.isPending`

Reply:
Sounds good. Done.